test/CreatorToken.t.sol

```diff
@@ -484,3 +497,51 @@ contract Pausing is CreatorTokenTest {
     vm.stopPrank();
   }
 }
+
+contract bondingCurveTests is CreatorTokenTest {
+  function test_BuyPriceIsCorrect(address _buyer) public {
+    vm.assume(
+      _buyer != address(0) && _buyer != address(creatorToken) && _buyer != creator
+        && _buyer != admin
+    );
+    uint256 buyNTokens = 5;
+
+    for (uint256 i = 0; i < buyNTokens; i++) {
+      uint256 originalBuyerBalance = creatorToken.balanceOf(_buyer);
+      (uint256 _tokenPrice, uint256 _creatorFee, uint256 _adminFee) = creatorToken.nextBuyPrice();
+      uint256 _totalPrice = _tokenPrice + _creatorFee + _adminFee;
+      require(_tokenPrice == bondingCurve.priceForTokenNumber(creatorToken.totalSupply() + 1));
+      require(_totalPrice == calculateTotalPrice(_tokenPrice));
```

Comment:
What this require statement does, in a roundabout way, is make sure the `calculateFees` helper returns the same thing as the `nextBuyPrice` returns. This isn't a bad thing to test, but:

* We should be using an `assert` statement for the same reason as above.
* We should do it more directly so it's clear what we're doing, i.e. by calling the `calculateFees` method in this test and asserting that each result matches the corresponding result from calling `nextBuyPrice`

Reply:
How does this look, https://github.com/showtime-xyz/creator-tokens/pull/14/commits/6a0ad6858e103fa6da6384492f59571f85783ba2

Reply:
Looks great! My only nit would be the last test (where you assert that total price and net proceeds line up with what is calculated in the test helpers) does not really get us anything. We're basically just checking that addition/subtraction in the test work. Once we've confirmed each of the constituent parts line up, there's no reason to test that adding/subtracting them together is the same.

Reply:
Gotcha. Removed them in the same commit as below: https://github.com/showtime-xyz/creator-tokens/pull/14/commits/0f60b38e1387e8109d0e5a6ca6f564d7a687b408